app/server.go

```diff
@@ -1454,6 +1465,93 @@ func doCheckWarnMetricStatus(a *App) {
 	}
 }
 
+func (s *Server) StopMetricsServer() {
+	s.metricsLock.Lock()
+	defer s.metricsLock.Unlock()
+
+	if s.metricsServer != nil {
+		s.metricsServer.Close()
+		s.Log.Info("Metrics and profiling server is stopping", mlog.String("address", *s.Config().MetricsSettings.ListenAddress))
+		s.metricsServer = nil
+	}
+}
+
+func (s *Server) HandleMetrics(route string, h http.Handler) {
+	if s.metricsRouter != nil {
```

Comment:
Is there a reason for this check? `HandleMetrics` is anyways called after `InitMetricsRouter`.

Reply:
It was for a safety measure since this method is called from ee repo as well. And sometimes it's hard to follow code from there. 0/5 to leave it.

Reply:
Sounds good. Let's keep it then.